tritium-registry/src/main/java/com/palantir/tritium/metrics/registry/MetricName.java

```diff
@@ -61,11 +40,12 @@
      * <p>
      * All tags and keys must be {@link Safe} to log.
      */
-    Map<String, String> safeTags();
+    @Value.NaturalOrder
+    NavigableMap<String, String> safeTags();
```

Comment:
Should we loosen this to SortedMap?

Reply:
I'd prefer to just leave this as a `NavigableMap` as the underlying impl generated by immutables is a `TreeMap` either way with `jdkOnly = true` (though we could remove that since we already rely on Guava `implementation` dependency, created optional https://github.com/palantir/tritium/pull/273 for that).

Reply:
I'd prefer the guava approach. We heavily rely on guava immutable collections, and they will avoid collection re-allocations when we attempt to re-wrap them.

I'm not sure any of the functionality provided by `NavigableMap` will be helpful for `safeTags`, where `SortedMap` provides clear advantages. Using the less specific type allows more freedom to change the implementation in the future. That said, I'm not actually aware of a `SortedMap` implementation which doesn't also implement `NavigableMap`, so I defer to your judgement.

Reply:
ok, you convinced me, went with `SortedMap` for now